src/effect.cpp

```diff
@@ -2,11 +2,168 @@
 #include "rng.h"
 #include "output.h"
 #include "player.h"
+
 #include <map>
 #include <sstream>
+#include <type_traits>
+#include <algorithm>
+#include <iterator>
 
 std::map<std::string, effect_type> effect_types;
 
+static std::string const eff_mods_base    = {"base_mods"};
+static std::string const eff_mode_scaling = {"scaling_mods"};
+
+static std::string const eff_arg_min        = {"min"};
+static std::string const eff_arg_max        = {"max"};
+static std::string const eff_arg_amount     = {"amount"};
+static std::string const eff_arg_min_val    = {"min_val"};
+static std::string const eff_arg_max_val    = {"max_val"};
+static std::string const eff_arg_chance_top = {"chance_top"};
+static std::string const eff_arg_chance_bot = {"chance_bot"};
+static std::string const eff_arg_tick       = {"tick"};
+
+static std::string const eff_type_str     {"STR"};
+static std::string const eff_type_dex   = {"DEX"};
+static std::string const eff_type_per   = {"PER"};
+static std::string const eff_type_int   = {"INT"};
+static std::string const eff_type_speed = {"SPEED"};
+
+static std::string const eff_type_pain    = {"PAIN"};
+static std::string const eff_type_hurt    = {"HURT"};
+static std::string const eff_type_sleep   = {"SLEEP"};
+static std::string const eff_type_pkill   = {"PKILL"};
+static std::string const eff_type_stim    = {"STIM"};
+static std::string const eff_type_health  = {"HEALTH"};
+static std::string const eff_type_h_mod   = {"H_MOD"};
+static std::string const eff_type_rad     = {"RAD"};
+static std::string const eff_type_hunger  = {"HUNGER"};
+static std::string const eff_type_thirst  = {"THIRST"};
+static std::string const eff_type_fatigue = {"FATIGUE"};
+static std::string const eff_type_cough   = {"COUGH"};
+static std::string const eff_type_vomit   = {"VOMIT"};
+
+/**
+ * clamp n to [lo, hi]
+ */
+template <typename T>
+inline T clamp(T const n, T const lo, T const hi) noexcept {
+    return (n < lo) ? lo :
+           (n > hi) ? hi :
+           n;
+}
+
+
+// This should probably be moved elsewhere if it is to be kept so that it can be more widely used
+namespace detail {
+
+template <typename To, typename From>
+struct select_result_type {
+    using type = typename std::conditional<
+        std::is_same<void, To>::value, From, To
+    >::type;
+};
+
+template <typename To, typename From>
+using select_result_type_t = typename select_result_type<To, From>::type;
+
+} //namespace detail
+
+/**
+ * do a round followed by a cast
+ *
+ * @tparam To optional cast type, the results will otherwise fallback to decltype(value)
+ */
+template <
+    typename To = void
+  , typename From
```

Comment:
It's not necessarily obvious from the style guidelines, but 100 characters per line implies we'd like to break lines only when that limit is reached, too many breaks is just as unreadable as not enough.
Also not supported by astyle is specifying where the comma goes in comma-delimited lists broken across lines, the answer is at the end of the line, which I honestly thought was the only viable choice.